mcstatus/protocol/connection.py

```diff
@@ -498,7 +499,12 @@ def __init__(self) -> None:
     def close(self) -> None:
         """Close :attr:`.socket`."""
         if self.socket is not None:  # If initialized
-            self.socket.shutdown(socket.SHUT_RDWR)
+            try:
+                self.socket.shutdown(socket.SHUT_RDWR)
+            except OSError as exception:  # Probably, the OS already closed the socket
```

Comment:
This is missleading, the socket wasn't closed by the OS at all here. All this error means is that there's nothing to shut down, as the socket isn't connected. Shutdown terminates the incomming and/or outcomming data to/from the socket gracefully, if it isn't connected, there's nothing to shut down, hence this error. But we still need to close the socket.

```suggestion
            except OSError as exception:  # Socket wasn't connected (nothing to shut down)
```

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/d38e931d2a108f6c70d94090fcdc1ab920102555..8bfbed48241416b24b4146ebbf8830df30a1ba78), thanks!